lib/workflow/stores/mongo.js

```diff
@@ -52,22 +52,26 @@ function mongoStoreFactory(waterline, Promise, Constants, Errors, assert, _) {
      * @param {Object} indexObject an object with keys that correspond to the mogo
      * collections on which to place indexes and values that are arrays of indexes
      * @example {
-     *          taskdependencies: [
-     *              {taskId: 1},
-     *              {graphId: 1},
-     *              {taskId: 1, graphId: 1}
-     *          ],
-     *          graphobjects: [
-     *              {instanceId: 1}
-     *          ]
+     *       taskdependencies: [
+     *           {index: {taskId: 1}, unique: true},
```

Comment:
An outstanding question might be does our mongo pass-through mean we still automatically create these indexes if they are in a waterline model, and also is there a way to make compound indexes via waterline, or does that stay with the mongo-specific implementation. If it all works with waterline, great, if not, perhaps we might want to at least implement our own abstraction at the model layer?